src/thread.cr

```diff
@@ -63,8 +68,10 @@ class Thread
 
       current_thread_id = LibC.pthread_self
 
-      current_thread = @@threads.find do |thread|
-        LibC.pthread_equal(thread.id, current_thread_id) != 0
+      current_thread = @@threads.each_key do |thread|
+        if LibC.pthread_equal(thread.id, current_thread_id) != 0
+          break thread
+        end
```

Comment:
Yes I mean `Enum#hash`, but I can go with `Signal#hash` instead. Whatever you prefer.